### ISSUE TITLE
How do we agree on things?

### ISSUE THREAD

**Issue by 3lvis** [bakkenbaeck/iOS-playbook#86]
We should have a framework for evaluating ideas and proposals. Steps to follow in each phase and ways to find a resolution when we don't agree on something unanimously. Finally, we should discuss what to do when someone from the team decides not to follow the playbook.

**Comment by elland** [comment]
_Since this is quite long, let me know if I should push this as a document so you can comment referencing lines etc._

This should be simple enough. We should stick to meaningful valid arguments (no false equivalencies, fallacies, strawmen arguments, and other cognitive biases. Since we're usually not aware of them, we should try to listen to others when they point that out). We each state our cases for a rule proposal. We should of course all be open to the arguments (given that we should only have good solid valid arguments on the table, after weeding out the invalid ones as they appear). Once that is done, it should be clear to everyone which arguments are the strongest. That should settle the matter.

Sometimes it's not that simple. People have a lot of idiosyncrasies and are usually very attached to a certain way of doing things. Most people are also not open to arguments or to changing their minds. The important thing to remember is that it's supposed to be a guideline we all agree on.

If we can't agree on a matter, we shouldn't codify it. Either because it's too heated, or people are too attached to weaker arguments instead of flocking to the stronger ones, or because some of us just have different priorities and what to one might be a very strong argument, to others it's not even something that registers as important. 

When that is the case, there's little point in arguing further, since the disagreement is at a much lower level. What do we do then? I think we should not ever steamroller our coworkers, by forcing it on them. I don't think we'll ever have the case where something that is concerning and hinders the flow of development, quality of code, readability and consistency would ever get to that point, because those are values we tend to all share. Disagreements of that sort tend to come only for small, sugary stuff. If that is the case and we can't reach an agreement, then we should try to generalise the rule further until it's something we can agree on or just not codify and leave it at the developer's discretion.

Of course there are debates that get heated and have no end in sight. Hello `spaces or tabs`? For these there's no way around. We have to pick one and move on. For those cases, we should probably go for the _law of least resistance_. What are the system or platform defaults? We should aim for those instead.

>  Finally, we should discuss what to do when someone from the team decides not to follow the playbook.

This should never happen. If this is happening, there's something else going wrong. This would probably be a symptom of steamrolling one's opinion over, either by imposition or by appealing to majorities. This should be avoided. Going for majority vote should be a last resort for critical guidelines or showstoppers. Like `tabs or spaces`. This is something that _needs_ to be decided, otherwise very pull-request or merge or diff will be a nightmare. Something less important, should either be generalised or not be codified at all.

For instance, we have a very specific git commit guideline imposition. This was discussed ad nauseam to no avail. A more generic higher level guideline we could all agree on was proposed, so we could move on. Instead the whole discussion was steamrolled and we kept the strict debated guideline anyway, completely overriding all the previous discussions and input. This should not be done. This creates attrition and doesn't add any real value to our team, just detracts from it.

**Comment by 3lvis** [comment]
Doing some spring cleaning, please open a new issue if you still want to do this.
